src/networkextension.cs

```diff
@@ -146,10 +161,26 @@ interface NEAppProxyFlow {
 	
 		[Export ("closeWriteWithError:")]
 		void CloseWrite ([NullAllowed] NSError error);
+
+		[NoWatch, NoTV, NoiOS, Mac (10,15,4)]
+		[Export ("setMetadata:")]
+		void SetMetadata (OS_nw_parameters parameters);
```

Comment:
Looks like they are unrelated, also swift has the [setMetadata](https://developer.apple.com/documentation/networkextension/neappproxyflow/3543399-setmetadata) method using this protocol [OS_nw_parameters](https://developer.apple.com/documentation/network/os_nw_parameters?language=objc) which is defined as `typedef NSObject<OS_nw_parameters> *nw_parameters_t;` so maybe this might be a better(?) binding?

```csharp
[NoWatch, NoTV, NoiOS, Mac (10,15,4)]
[Export ("setMetadata:")]
void SetMetadata (OS_nw_parameters parameters);
```

@spouliot what do you think?

Reply:
Or just expose `OS_nw_parameters` protocol with a better C# name and just bind it as` IOS_nw_parameters` 🤔

Reply:
@mandel-macaque I would also like your take on this one since you bound `Network` :)

Reply:
hmm... we have API that returns `nw_parameters_t` instances

https://developer.apple.com/documentation/network/2976839-nw_parameters_create_secure_tcp?language=objc

so why not use `NWParameters` ?

It's (again) mixing `NSObject`/`INativeObject`, which requires some additional code glue, but it's the right thing to do, i.e.

* `void SetMetadata (NSObject parameters);` would not work with an instance of `NWParameters`

* `void SetMetadata (IntPtr parameters);` would _work_ with an instance of `NWParameters`, but would be _needlessly_ ugly `SetMetadata (x.Handle);` and error prone (no type check).

Reply:
@spouliot because I am dumb and missed the `NWParameters` parameters class 🙈, I'll use that instead